### ISSUE TITLE
Meshroom can't estimate my second camera's images, but why does it work well alone? [question]

### ISSUE THREAD

**Issue by JohannSuarez** [alicevision/Meshroom#2132]
I have a DJI Mini 2 and a Canon EOS M50 Mark 1 that I both used for taking photos of a subject.
The intention of using two cameras is that the drone is for scanning the whole body, while the Canon is for taking more high-res pictures of the subject's head. When I run the Canon M50's images alone, they are estimated reasonably well with some images failing to be estimated

Canon M50:
![m50](https://github.com/alicevision/Meshroom/assets/20419658/5af361d6-5093-46eb-824b-81e8d865dc41)

I ran the drone's images alone as well, and all frames were esimated without failure, as you can see here.

Drone images:
![drone](https://github.com/alicevision/Meshroom/assets/20419658/8219ea4f-8872-4e48-aeb6-afe1bab75ae4)

But when I run both image sets through the pipeline at the same time, not a single one of the drone images get estimated. Here's a pic:
![only_the_m50](https://github.com/alicevision/Meshroom/assets/20419658/1a4c009f-0589-4457-84fc-22cdf2642246)

Why can't I use both image sets simultaneously? They all have their focal lengths in their metadata. The drone images are 4mm and the M50 images are 30mm.

 - OS: Win 10
 - Python version: 3.7.3 
 - Meshroom version: Release 2023.2.0

**Comment by msanta** [comment]
The close up photos have a shallow depth of field, so the background is not very sharp. Now if you try to add the distance photos it could be that there are not enough matches found in the background to tie the two sets together. The person themselves is not covering enough of the image area to allow robust matching to be obtained.

You could try to take some photos between the closeups and distance to act a bridge between the two sets.

The other issue you will have is that the subject will be moving, even a tiny bit, between photos.

**Comment by natowi** [comment]
I would try something like this:
Use the drone dataset reconstruction with the 15 matches from the close-up.
Then do a second reconstruction pipeline with the close-up images. Then use a common image that is used in both dataset to align both SfMs to each other (SfMAlignment) node. Use a matching method: 
- Alignment Method:
- from_cameras_viewid: Align cameras with same view Id
- from_cameras_poseid: Align cameras with same pose Id
- from_cameras_filepath: Align cameras with a filepath matching, using ‘fileMatchingPattern’

If the close-up reconstruction is not so good, you could try [masking ](https://github.com/alicevision/Meshroom/wiki/New-Features-in-Meshroom-2023.1#1-image-masking) out the background (best generate masks for the background using some other tool).

**Comment by tak-ho** [comment]
> I would try something like this: Use the drone dataset reconstruction with the 15 matches from the close-up. Then do a second reconstruction pipeline with the close-up images. Use a common image that is matched in both datasets to align both SfMs to each other (SfMAlignment) node.
> 
> * Alignment Method:
> * from_cameras_viewid: Align cameras with same view Id
> * from_cameras_poseid: Align cameras with same pose Id
> * from_cameras_filepath: Align cameras with a filepath matching, using ‘fileMatchingPattern’
> 
> If the close-up reconstruction is not so good, you could try [masking ](https://github.com/alicevision/Meshroom/wiki/New-Features-in-Meshroom-2023.1#1-image-masking) out the background (best generate masks for the background using some other tool).

I just read this thread that may related to my situation too #2157 
Is this Augment Reconstruction? Do I need to enable "Lock Scene Previously Reconstructed"

Thanks!

**Comment by natowi** [comment]
No, this (SfMAlignment) is not equal to Augment Reconstruction. SfMAlignment aligns two individual reconstructions based on common information like same camera filepath.

**Comment by tak-ho** [comment]
Thanks! @natowi then it seems its not working in my case...
